sky/optimizer.py

```diff
@@ -897,7 +907,8 @@ def _make_launchables_for_valid_region_zones(
     launchables = []
     regions = launchable_resources.get_valid_regions_for_launchable()
     for region in regions:
-        if launchable_resources.use_spot and region.zones is not None:
+        if launchable_resources.use_spot and region.zones is not None \
+            or isinstance(launchable_resources.cloud, clouds.GCP):
```

Comment:
nit:  we prefer using `()` than backslash. : )
```suggestion
        if (launchable_resources.use_spot and region.zones is not None
            or isinstance(launchable_resources.cloud, clouds.GCP)):
```